packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -139,52 +140,59 @@ function prepareArrayContentForHydration(
 	content: readonly MapTree[],
 	forest: IForestSubscription,
 ): void {
-	const proxies: RootedProxyPaths[] = [];
-	for (const [i, item] of content.entries()) {
-		proxies.push({
+	const proxyPaths: RootedProxyPaths[] = [];
+	for (const item of content) {
+		const proxyPath: RootedProxyPaths = {
 			rootPath: {
 				parent: undefined,
 				parentField: EmptyKey,
 				parentIndex: 0,
 			},
 			proxyPaths: [],
-		});
-		// Non null asserting here because we are iterating over content and pushing into proxies for every content
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		walkMapTree(item, proxies[i]!.rootPath, (p, proxy) => {
-			// Non null asserting here because we are iterating over content and pushing into proxies for every content
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			proxies[i]!.proxyPaths.push({ path: p, proxy });
+		};
+		proxyPaths.push(proxyPath);
+		walkMapTree(item, proxyPath.rootPath, (p, proxy) => {
+			proxyPath.proxyPaths.push({ path: p, proxy });
 		});
 	}
 
-	bindProxies(proxies, forest);
+	bindProxies(proxyPaths, forest);
 }
 
 function walkMapTree(
 	mapTree: MapTree,
 	path: UpPath,
 	onVisitTreeNode: (path: UpPath, treeNode: TreeNode) => void,
 ): void {
-	const mapTreeNode = tryGetMapTreeNode(mapTree);
-	if (mapTreeNode !== undefined) {
-		const treeNode = tryGetCachedTreeNode(mapTreeNode);
-		if (treeNode !== undefined) {
-			onVisitTreeNode(path, treeNode);
-		}
+	if (tryGetMapTreeNode(mapTree)?.parentField.parent.parent !== undefined) {
```

Comment:
why two levels of parent?

Reply:
`parentField` gets you a `{ parent: Field, ... }`, so `parentField.parent` gets you a `Field`. So if you actually want to check the parent _node_, you must do `parentField.parent.parent`. So it is only one parent up. Also, we can't simply check the parent field because the `parentField` (and `parentField.parent`) is always present (all nodes have a parent field, though it may be the detached field). But the field may not have a parent node, if and only if it is the detached field.